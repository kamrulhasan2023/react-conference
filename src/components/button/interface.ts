import React from 'react';
import { IconNames } from './../icons/interface';

export const buttonSizes = ['lg', 'md', 'sm', 'xs'] as const;
type Size = (typeof buttonSizes)[number];

export interface IFillButton {
  label: string;
  colorScheme: ColorScheme;
  size: Size;
  clickHandler: (
    e: React.MouseEvent<HTMLElement> | React.FormEvent<HTMLFormElement>
  ) => void;
  hoverColorScheme?: ColorScheme;
  iconName?: IconNames;
  iconIsLeft?: boolean;
  isDisabled?: boolean;
  variant?: 'fill';
  type?: 'button';
}

export interface IFillSubmitButton
  extends Omit<IFillButton, 'clickHandler' | 'type'> {
  type: 'submit';
}

export interface IFillRedirectButton
  extends Omit<IFillButton, 'clickHandler' | 'isDisabled'> {
  href?: string;
  isOpenNewTab: boolean;
}

export interface IOutlineButton
  extends Omit<IFillButton, 'colorScheme' | 'hoverColorScheme' | 'variant'> {
  colorScheme: OutlineColorScheme;
  hoverColorScheme: OutlineColorScheme;
  variant: 'outline';
}

export interface IOutlineSubmitButton
  extends Omit<IOutlineButton, 'clickHandler' | 'type'> {
  type: 'submit';
}

export interface IOutlineRedirectButton
  extends Omit<IOutlineButton, 'clickHandler' | 'isDisabled'> {
  href?: string;
  isOpenNewTab: boolean;
}

export type IBaseButton =
  | Omit<IOutlineButton, 'clickHandler' | 'type'>
  | Omit<IFillButton, 'clickHandler' | 'type'>;

export const buttonColorSchemes = [
  'bg-primary-text-white',
  'bg-success-text-white',
  'bg-black-text-white',
  'bg-white-text-black',
] as const;
type ColorScheme = (typeof buttonColorSchemes)[number];

export const outlineButtonColorSchemes = [
  'bg-transparent-text-primary',
  'bg-transparent-text-success',
] as const;
type OutlineColorScheme = (typeof outlineButtonColorSchemes)[number];
