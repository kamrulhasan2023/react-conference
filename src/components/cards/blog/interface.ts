import { IDimension } from 'src/components/custom-image/interface';

export interface IBlogCard {
  href: string;
  image: {
    src: string;
    alt: string;
    lqip: string;
  };
  title: string;
  description: string;
  tags: {
    bgColor: { hex: string };
    fontColor: { hex: string };
    title: string;
  }[];
  isFeatured: boolean;
}

export const BLOG_IMAGE_DIMENSION: IDimension = {
  width: 960,
  height: 720,
};
