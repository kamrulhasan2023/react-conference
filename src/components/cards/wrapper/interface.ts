export interface ICardWrapper {
  cardShadow: ICardShadow;
  hoverCardShadow?: ICardShadow;
  cardBorderRadius: ICardBorderRadius;
}

export const cardWrapperShadow = ['shadow-one', 'shadow-none'] as const;
type ICardShadow = (typeof cardWrapperShadow)[number];

export const cardWrapperBorderRadius = [
  'radius-one',
  'radius-two',
  'radius-none',
] as const;
type ICardBorderRadius = (typeof cardWrapperBorderRadius)[number];
