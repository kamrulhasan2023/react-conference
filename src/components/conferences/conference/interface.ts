export interface IConference{
    id ? : string | number,
    title: string;
    description:string;
    startDate: string;
    count:number,
    isLastItem? : boolean;
}