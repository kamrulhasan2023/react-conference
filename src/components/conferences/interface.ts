export interface IConferences  {
    title : string;
    conferences :Array<{
        id: string | number;
        name: string;
        slogan:string;
        startDate: string;
    }>;
}