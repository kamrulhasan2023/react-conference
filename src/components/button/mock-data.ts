import {
  IFillButton,
  IFillRedirectButton,
  IOutlineButton,
  IOutlineRedirectButton,
} from './interface';

export const buttonProps: IFillButton = {
  label: 'Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  clickHandler: () => {},
  variant: 'fill',
};

export const redirectButtonProps: IFillRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-success-text-white',
  size: 'lg',
  variant: 'fill',
};

export const outlineButtonProps: IOutlineButton = {
  label: 'Outline Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  clickHandler: () => {},
  variant: 'outline',
};

export const redirectOutlineButtonProps: IOutlineRedirectButton = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-success',
  size: 'lg',
  variant: 'outline',
};
