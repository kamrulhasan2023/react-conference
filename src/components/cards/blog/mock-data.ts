import { IBlogCard } from './interface';

export const blogCardProps: IBlogCard = {
  href: 'https://www.lemonhive.com/',
  image: {
    src: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
    alt: 'dummy image 2',
    lqip: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
  },
  title: 'Headless Website Development',
  description:
    'Headless solutions built with Sanity.io, Strapi, Prismic, Contentful and others with front-end frameworks such as Next.js, Gatsby, & Nuxt.',
  tags: [
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Reactjs' },
    { fontColor: { hex: 'black' }, bgColor: { hex: 'red' }, title: 'Nextjs' },
  ],
  isFeatured: false,
};
