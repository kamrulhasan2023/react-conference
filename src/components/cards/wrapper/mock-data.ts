import { ICardWrapper } from './interface';

export const cardWrapperProps: ICardWrapper = {
  cardBorderRadius: 'radius-none',
  cardShadow: 'shadow-one',
};
