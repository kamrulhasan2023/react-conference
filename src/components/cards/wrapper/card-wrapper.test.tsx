import { render } from '@testing-library/react';
import CardWrapper from '.';
import { cardWrapperProps } from './mock-data';
import { cardWrapperBorderRadius, cardWrapperShadow } from './interface';

it('renders card wrapper correctly', () => {
  const { getByRole } = render(
    <CardWrapper {...cardWrapperProps}>
      <h2>Card Wrapper</h2>
    </CardWrapper>
  );
  const element = getByRole('presentation');
  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});

describe.each(cardWrapperShadow)(
  'Given card wrapper box shadow: %s',
  (boxShadow) => {
    it(`Expect card wrapper to have class ${boxShadow}`, () => {
      const { getByRole } = render(
        <CardWrapper {...cardWrapperProps} cardShadow={boxShadow}>
          <h2>CardWrapper</h2>
        </CardWrapper>
      );
      const element = getByRole('presentation');
      expect(element).toHaveClass(boxShadow);
    });
  }
);
describe.each(cardWrapperBorderRadius)(
  'Given card wrapper shadow: %s',
  (borderRadius) => {
    it(`Expect card wrapper to have class ${borderRadius}`, () => {
      const { getByRole } = render(
        <CardWrapper {...cardWrapperProps} cardBorderRadius={borderRadius}>
          <h2>CardWrapper</h2>
        </CardWrapper>
      );
      const element = getByRole('presentation');
      expect(element).toHaveClass(borderRadius);
    });
  }
);
