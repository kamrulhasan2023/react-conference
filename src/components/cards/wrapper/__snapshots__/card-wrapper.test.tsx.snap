// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders card wrapper correctly 1`] = `
<div
  class="relative overflow-hidden h-auto card shadow-one radius-none"
  role="presentation"
>
  <h2>
    Card Wrapper
  </h2>
</div>
`;
